### PR TITLE
fix: HTTPRoute status only shows one parent when targeting multiple Gateways from different GatewayClasses

### PR DESCRIPTION
**Pull request by zhaohuabing**
Fixes: https://github.com/envoyproxy/gateway/issues/4264

Release Notes: Yes

### CODE REVIEW

File: internal/provider/kubernetes/status.go [hunk 1]
```diff
@@ -469,6 +469,29 @@ func (r *gatewayAPIReconciler) subscribeAndUpdateStatus(ctx context.Context, ext
...
 
+// mergeRouteParentStatus merges the old and new RouteParentStatus.
+// This is needed because the RouteParentStatus doesn't support strategic merge patch yet.
+func mergeRouteParentStatus(ns string, old, new []gwapiv1.RouteParentStatus) []gwapiv1.RouteParentStatus {
+	merged := make([]gwapiv1.RouteParentStatus, len(old))
+	_ = copy(merged, old)
+	for _, parent := range new {
+		found := -1
+		for i, existing := range old {
+			if parent.ParentRef.Name == existing.ParentRef.Name &&
```
Comment (arkodg) [inline internal/provider/kubernetes/status.go:480]: also && for `parent.ControllerName` && `parent.ParentRef.Kind`

File: internal/provider/kubernetes/status.go [hunk 2]
```diff
@@ -469,6 +469,29 @@ func (r *gatewayAPIReconciler) subscribeAndUpdateStatus(ctx context.Context, ext
...
+				utils.NamespaceDerefOr(parent.ParentRef.Namespace, ns) == utils.NamespaceDerefOr(existing.ParentRef.Namespace, ns) {
```
Comment (arkodg) [inline internal/provider/kubernetes/status.go:481]: "k8s.io/utils/ptr"
we've updated a lot of the code in the repo to use `ptr.Deref(parent.ParentRef.Namespace, ns)` instead

File: internal/provider/kubernetes/status.go (hunk 1 above)
Comment (arkodg) [reply internal/provider/kubernetes/status.go:480]: do we also need to check for sectionName / listener name if non nil ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**